pkg/api/index.go

```diff
@@ -422,11 +422,11 @@ func (hs *HTTPServer) buildDashboardNavLinks(c *models.ReqContext, hasEditPerm b
 			Text: "Divider", Divider: true, Id: "divider", HideFromTabs: true,
 		})
 		dashboardChildNavs = append(dashboardChildNavs, &dtos.NavLink{
-			Text: "New dashboard", Icon: "plus", Url: hs.Cfg.AppSubURL + "/dashboard/new", HideFromTabs: true,
+			Text: "New dashboard", Icon: "plus", Url: hs.Cfg.AppSubURL + "/dashboard/new", HideFromTabs: true, Id: "new-dashboard",
```

Comment:
This feels a bit weird to me. I might have missed something but why are we not applying the translation here, in the backend, so the `Text` will have the proper translation instead of sending out the `Id` and do it in the frontend?

Reply:
It's a fair point. That would then require localising the backend, which is something we're trying to avoid doing just for simplicity.

Backend is out of scope for localisation any time soon, and it's just accepted that some things like error messages that come from the backend won't be translated. Although we would/could/should start to use more standardised error handling and error codes, and the frontend can apply its own (localised) error messages.

Reply:
That would be a mistake, according to me. What is the reason behind that decision?

I'm not sure it is making things simpler. You will most likely need to have this working fullstack in a near future for alerting, emails etc that will be sent out from the backend. Or that is what have happen on other companies I have been at (so this might be valid for Grafana).

Have you thought about letting the backend the translations and provide them to the frontend for the currently selected locale? Then you will be able to use them within the whole stack.

Reply:
Sorry - less "simplicity" and more "not in scope _at this moment_."

The majority of things that need to be translated are in the frontend, so that's where this effort is starting. I agree that long term more things from the backend that don't even touch the frontend will need to be translated (like the things you mentioned - emails and alert notifs), but we've already got quite a lot to learn as it is so we're starting with a well defined reasonably limited scope and iterating from there.

Tech things we want to learn right now are:
 - how do we mark up things in the frontend?
   - _can_ we use this same approach to translate parts of the frontend that come from the backend, like Navigation?
 - what localisation provider stack will we use?
 - how do we integrate a full translation workflow into our development workflow? 

:v:

Reply:
Got it and I think it is a valid approach! At the same time I feel like this use case would be a pretty good validator to prove that the current solution will be applicable on both frontend and backend to minimize the risk of needing to re-think the solution when starting to implement it in the backend.